src/OrleansProviders/Streams/Generator/GeneratorPooledCache.cs

```diff
@@ -58,13 +60,65 @@ public bool Equals(CachedMessage cachedMessage, IStreamIdentity streamIdentity)
             }
         }
 
+        private class ExplicitEvictionStrategy : IEvictionStrategy<CachedMessage>
+        {
+            private FixedSizeBuffer currentBuffer;
+            public IPurgeObservable<CachedMessage> PurgeObservable { set; private get; }
+
+            public Action<CachedMessage?, CachedMessage?> OnPurged { get; set; }
+
+            //Explicitly purge all messages in purgeRequestBlock
+            public void PerformPurge(DateTime utcNow, IDisposable purgeRequest)
+            {
+                //if the cache is empty, then nothing to purge, return
+                if (this.PurgeObservable.ItemCount == 0)
+                    return;
+                var itemCountBeforePurge = this.PurgeObservable.ItemCount;
```

Comment:
yes, I guess I just remember to clean that logic in EHCache. forgot other cache.